### PR TITLE
Validate config values against the config schema

### PR DESCRIPTION
**Pull request by Kami**
This pull request adds `--register-validate-configs` flag to the `st2-register-content` script.

If this flag is available, new-style config values are validated against the config schema if config schema is available. As mentioned in the past, this works along the whole gradual / optional typing story so the validation is not enabled by default right now, but a user can opt-in using a flag mentioned above.

It's also to important to note that only "static" config values are validated during the registration phase. Dynamic ones are resolved during run-time so those can only be validated during run-time (when a sensor is instantiated for sensors and when action is ran for actions). That's not something I will do in this PR, but I will introduce support for that at some point in the future (baby steps).

Usage:

```bash
st2-register-content --config-file conf/st2.dev.conf --register-configs --register-validate-configs
```

Example output on failure:

```bash
2016-07-29 14:20:50,296 INFO [-] Connecting to database "st2" @ "0.0.0.0:27017" as user "None".
2016-07-29 14:20:50,410 INFO [-] =========================================================
2016-07-29 14:20:50,411 INFO [-] ############## Registering configs ######################
2016-07-29 14:20:50,411 INFO [-] =========================================================
2016-07-29 14:20:50,975 WARNING [-] Failed to register configs: Failed validating config for pack "aws": 1 is not of type u'string'

Failed validating u'type' in schema['properties'][u'region']:
    {u'default': u'us-east-1',
     u'description': u'AWS region to use.',
     u'required': True,
     u'type': u'string'}

On instance[u'region']:
    1
Traceback (most recent call last):
  File "./st2common/bin/st2-register-content", line 22, in <module>
    sys.exit(content_loader.main(sys.argv[1:]))
  File "/data/stanley/st2common/st2common/content/bootstrap.py", line 359, in main
    register_content()
  File "/data/stanley/st2common/st2common/content/bootstrap.py", line 343, in register_content
    register_configs()
  File "/data/stanley/st2common/st2common/content/bootstrap.py", line 307, in register_configs
    raise e
jsonschema.exceptions.ValidationError: Failed validating config for pack "aws": 1 is not of type u'string'

Failed validating u'type' in schema['properties'][u'region']:
    {u'default': u'us-east-1',
     u'description': u'AWS region to use.',
     u'required': True,
     u'type': u'string'}

On instance[u'region']:
    1
```

## TODO

- [x] Tests
- [x] Documentation updates - https://github.com/StackStorm/st2/pull/2840

### CODE REVIEW

File: st2common/st2common/content/bootstrap.py [hunk 1]
```diff
@@ -56,10 +56,15 @@ def register_opts():
         cfg.BoolOpt('policies', default=False, help='Register policies.'),
         cfg.BoolOpt('configs', default=False, help='Register and load pack configs.'),
 
+        # Config related options
+        cfg.BoolOpt('validate-configs', default=False,
```
Comment (Kami) [inline st2common/st2common/content/bootstrap.py]: I'm also open to a better flag name - suggestions welcome.

**Comment by Kami** [comment]
Docs at https://github.com/StackStorm/st2/pull/2840

File: st2common/st2common/content/bootstrap.py (hunk 1 above)
Comment (lakshmi-kannan) [inline st2common/st2common/content/bootstrap.py]: I've no idea why we want a flag for this? If there is a schema, we should by default validate and throw an error. We now have flags for skip failure or abort (default). I think those should be enough.

**Comment by lakshmi-kannan** [comment]
I really like the validation change but i think the flag to turn on the validation is excessive IMO. We should try and reduce the number of flags and not add more :P.

**Comment by Kami** [comment]
I'm fine with making validation mandatory if config schema exists.

**Comment by Kami** [comment]
I will make the changes tomorrow - turn it on by default and remove the flag.

**Pull request merged**